### PR TITLE
Add CTD pages in ES, IT, and PL [fix #14264]

### LINKED ISSUE #14264: Add CTD pages in ES, IT, and PL
### **_NOTE: we will be changing the DE URL before completing this request_**. https://github.com/mozilla/bedrock/issues/14266

The CTD landing page is now being expanded to ES, IT, and PL. 

Info and resources (Mozilla-only):

- **[Copy doc]**(https://docs.google.com/spreadsheets/d/1KF9hfxONY7Dccep7InJEECLPJkFgDZzddJmSXsA7Gj8/edit#gid=764047959) (each language in a different tab).

- **Figma file**(https://www.figma.com/file/7QLwJIytyYUIL4NDkZS16e/Firefox_DMA?type=design&node-id=574%3A11882&mode=design&t=crF1R7ekYy8iY68j-1), like the metadata imgs and translated lockups.

- **[Asana task](https://app.asana.com/0/1204604936721242/1206403557251908/f)**

New translations will reside at /firefox/challenge-the-default/ (but the German page remains at /firefox/ for now).

### PR DESCRIPTION
**Pull request by craigcook**
## One-line summary
Adds Spanish, Italian, and Polish versions of the Challenge the Default page.

This also moves the German page to /de/challenge-the-default/ and sets /firefox/ back to the standard page.

## Issue / Bugzilla link
#14264 
#14266 

## Testing
http://localhost:8000/de/firefox/ - should be the regular Firefox page
http://localhost:8000/de/firefox/challenge-the-default/
http://localhost:8000/es-ES/firefox/challenge-the-default/
http://localhost:8000/it/firefox/challenge-the-default/
http://localhost:8000/pl/firefox/challenge-the-default/

### CODE REVIEW

File: bedrock/firefox/templates/firefox/challenge-the-default/landing-base.html [hunk 1]
```diff
@@ -362,12 +362,14 @@ <h3>{{ cred_private_title }}</h3>
         </summary>
         <p>{{ cred_private_body }}</p>
       </details>
+    {% if cred_ethical_title %}
```
Comment (craigcook) [inline bedrock/firefox/templates/firefox/challenge-the-default/landing-base.html:365]: I had no copy for this bit for Polish, not sure if it was intentionally omitted or just missed in translation.

File: bedrock/firefox/templates/firefox/challenge-the-default/langs/landing.es-ES.html
Comment (reemhamz) [inline bedrock/firefox/templates/firefox/challenge-the-default/langs/landing.es-ES.html:1]: There's a missing translation for the easter egg that pops up from under the header "window" when x'd out -- it's still in French:
Screenshot 2024-03-13 at 10 43 01 AM

**Review by reemhamz — APPROVED** [review]
Just one nit (blocking), but other than that, this was a pretty clean PR. r+wc 💯

File: bedrock/firefox/templates/firefox/challenge-the-default/langs/landing.es-ES.html
Comment (craigcook) [reply bedrock/firefox/templates/firefox/challenge-the-default/langs/landing.es-ES.html:1]: oops missed one!

**Pull request merged**